ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -1062,53 +1082,95 @@ namespace OrdinalAxis {
+                ordinalPositions[ordinalLength - 1] >= val
+            ) {
+                ordinalIndex = getIndexInArray(ordinalPositions, val);
+                // final return value is based on ordinalIndex
             } else {
 
-                let ordinalLength = ordinalPositions.length,
-                    i,
-                    distance,
-                    ordinalIndex;
+                if (!extendedOrdinalPositions) {
+                    extendedOrdinalPositions =
+                        ordinal.getExtendedPositions &&
+                        ordinal.getExtendedPositions();
+                    ordinal.extendedOrdinalPositions = extendedOrdinalPositions;
+                }
+                if (!(extendedOrdinalPositions && extendedOrdinalPositions.length)) {
 
-                // first look for an exact match in the ordinalpositions array
-                i = ordinalLength;
-                while (i--) {
-                    if (ordinalPositions[i] === val) {
-                        ordinalIndex = i;
-                        break;
-                    }
+                    return val;
                 }
 
-                // if that failed, find the intermediate position between the
-                // two nearest values
-                i = ordinalLength - 1;
-                while (i--) {
-                    if (val > ordinalPositions[i] || i === 0) { // interpolate
-                        // something between 0 and 1
-                        distance = (val - ordinalPositions[i]) /
-                            (ordinalPositions[i + 1] - ordinalPositions[i]);
-                        ordinalIndex = i + distance;
-                        break;
+                const length = extendedOrdinalPositions.length;
+
+                if (!slope) {
+                    slope =
+                        (extendedOrdinalPositions[length - 1] -
+                            extendedOrdinalPositions[0]) /
+                        length;
+                }
+                // OriginalPointReference is equal to the index of
+                // first point of ordinalPositions in extendedOrdinalPositions.
+
+                let originalPositionsReference = getIndexInArray(
+                    extendedOrdinalPositions,
+                    ordinalPositions[0]
+                );
+
+                // If the searched value is outside the visiblePlotArea,
+                // check if it is inside extendedOrdinalPositions.
+                if (
+                    val >= extendedOrdinalPositions[0] &&
+                    val <=
+                        extendedOrdinalPositions[
+                            extendedOrdinalPositions.length - 1
+                        ]
+                ) {
+                    // Return Value
+                    ordinalIndex = getIndexInArray(extendedOrdinalPositions, val) - originalPositionsReference;
+                } else {
+                    // since ordinal.slope is the average distance between 2
```

Comment:
In general, I would skip this, but since there are some other changes, could you start comment from the capital letter? ;)